hack/generator/pkg/astmodel/errored_type.go

```diff
@@ -20,20 +20,24 @@ type ErroredType struct {
 	warnings []string
 }
 
-func MakeErroredType(t Type, errors []string, warnings []string) ErroredType {
-	return ErroredType{
+var _ Type = &ErroredType{}
+
+func MakeErroredType(t Type, errors []string, warnings []string) *ErroredType {
+	result := &ErroredType{
 		inner:    nil,
 		errors:   errors,
 		warnings: warnings,
-	}.WithType(t) // using WithType ensures warnings and errors get merged if needed
+	}
+
+	return result.WithType(t) // using WithType ensures warnings and errors get merged if needed
 }
 
-func (e ErroredType) InnerType() Type {
+func (e *ErroredType) InnerType() Type {
```

Comment:
Suggest being consistent with receiver name
```suggestion
func (errored *ErroredType) InnerType() Type {
```

Reply:
FWIW I think @babbageclunk had argued against long (> a few letters) receiver names in the past as non-idiomatic.

Reply:
I don't mind short or long (within reason) but all the other methods in this file are using `errored` already, hence the suggestion. Could rename them all to `e` or `et`.